### DISCUSSION TITLE
LOVR Plugins and Android

### DISCUSSION THREAD

**Discussion by CounterParry** [bjornbytes/lovr#768] [Q&A] [answered]
I was wanting to use LOVR on android with a plugin.   But I see this may cause a signing issue. 

> Prebuilt plugin libraries can also be copied next to the lovr executable. However, this won't work easily for Android builds, since the APK requires re-signing if it is modified.

It seems all I would need to do is resign the APK after adding the plugin.  Am I missing something? Has anyone done this?

**Comment by bjornbytes** [comment, marked as answer]
If you are building with CMake, you can add the plugins to the `plugins` folder and LÖVR will build them along with the rest of the project and sign the final APK for you.  If you've managed to get an Android CMake build working, that's probably the easiest way to do plugins on Android.

If you're trying to add plugins to an existing APK, it's probably possible but I haven't heard of anyone doing it yet.  The process would be something like this:

- Build the plugin for Android (aka ARM64/aarch64).
- Add the file to the lib/arm64-v8a folder of the APK:
  - Unzip the APK and add the plugin file to `lib/arm64-v8a`.  Then rezip.
  - Use aapt: `aapt add lovr.apk lib/arm64-v8a/libmyplugin.so`
- Run `zipalign` on the APK again: `zipalign -f -p 4 lovr.apk lovr.apk.aligned`
- Sign the APK: `apksigner sign --ks path/to/key.keystore --ks-pass pass:hunter2 --in lovr.apk.aligned --out lovr.apk`

Good luck!  Android tooling is not for the faint of heart
